x/mint/keeper/query_server.go

```diff
@@ -46,10 +54,138 @@ func (q queryServer) Inflation(ctx context.Context, _ *types.QueryInflationReque
 		Mul(math.NewIntFromUint64(blocksPerMonth)).
 		Mul(math.NewInt(12)).
 		ToLegacyDec()
-	totalSupply := q.k.GetTotalCurrTokenSupply(ctx).Amount.ToLegacyDec()
-	inflation := EmissionPerYearAtCurrentBlockEmissionRate.Quo(totalSupply).MulInt64(100)
+	circulatingSupply, err := GetCirculatingSupply(ctx, q.k, moduleParams, blockHeight, blocksPerMonth)
+	if err != nil {
+		return nil, err
+	}
+	inflation := EmissionPerYearAtCurrentBlockEmissionRate.QuoInt(circulatingSupply).MulInt64(100)
 	ret := types.QueryInflationResponse{
 		Inflation: inflation,
 	}
 	return &ret, nil
 }
+
+// mint and inflation emission rate endpoint
+// nice way to access live chain data
+func (q queryServer) EmissionInfo(ctx context.Context, _ *types.QueryEmissionInfoRequest) (*types.QueryEmissionInfoResponse, error) {
+	moduleParams, err := q.k.Params.Get(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get module params")
+	}
+
+	ecosystemBalance, err := q.k.GetEcosystemBalance(ctx, moduleParams.MintDenom)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get ecosystem balance")
+	}
+
+	previousBlockEmission, err := q.k.PreviousBlockEmission.Get(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get previous block emission")
+	}
+
+	ecosystemMintSupplyRemaining, err := q.k.GetEcosystemMintSupplyRemaining(ctx, moduleParams)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get ecosystem mint supply remaining")
+	}
+
+	blocksPerMonth, err := q.k.GetParamsBlocksPerMonth(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get blocks per month")
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	blockHeight := uint64(sdkCtx.BlockHeight())
+	numberOfRecalcs := blockHeight / blocksPerMonth
+	blockHeightTarget_e_i_LastCalculated := numberOfRecalcs*blocksPerMonth + 1
+	blockHeightTarget_e_i_Next := blockHeightTarget_e_i_LastCalculated + blocksPerMonth
+
+	networkStakedTokens, err := GetNumStakedTokens(ctx, q.k)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get number of staked tokens")
+	}
+	lockedVestingTokensTotal, lockedVestingTokensPreseed,
+		lockedVestingTokensSeed, lockedVestingTokensTeam := GetLockedVestingTokens(
+		blocksPerMonth,
+		math.NewIntFromUint64(blockHeight),
+		moduleParams)
+	ecosystemLocked := ecosystemBalance.Add(ecosystemMintSupplyRemaining)
+	circulatingSupply := moduleParams.MaxSupply.Sub(lockedVestingTokensTotal).Sub(ecosystemLocked)
```

Comment:
maybe we could call the function to get the circulating supply instead?

Reply:
yeah idk. It's less efficient if we do that because the circulating supply function calls variables that we then use later so it's two lookups instead of one for the same data, but on the other hand the code looks cleaner to read, is one less place to have to worry about differences, etc

Reply:
oh sorry, thought you were asking about in abci.go `GetEmissionPerMonth` - I'm still on the fence there

for query server yes makes sense

Reply:
few look ups vs abstraction for somewhat involved calc -- I lean abstraction